src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -381,6 +393,49 @@ async def debug_python_test(field_set: PythonTestFieldSet) -> TestDebugRequest:
     )
 
 
+@rule(desc="Set up debugpy to run an interactive Pytest session", level=LogLevel.DEBUG)
+async def debugpy_python_test(
+    field_set: PythonTestFieldSet,
+    debugpy: DebugPy,
+    pytest: PyTest,
+) -> TestDebugAdapterRequest:
+    debugpy_pex = await Get(Pex, PexRequest, debugpy.to_pex_request())
+    if pytest.main.spec != "pytest":
+        logger.warn(
```

Comment:
Should be `.warning`. `.warn` is deprecated

Reply:
https://github.com/pantsbuild/pants/pull/15837